### PR TITLE
Switch between crypto and credit card payment

### PR DESCRIPTION
**Pull request by cnasc**
# Description

This PR allows crypto users (i.e., not people with Unlock user accounts) to switch between paying with a credit card and paying with the lock's native currency.

This is nearly complete, but needs a good pass of manual testing and probably a few small tweaks to rendering logic.

image

image

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
So excited by this!

**Comment by cnasc** [comment]
Before this is ready to go it will require some changes to the purchase process on the fiatlocks component, which currently assumes it will use the unlock provider

File: unlock-app/src/components/content/CheckoutContent.tsx [hunk 1]
```diff
@@ -156,6 +158,19 @@ export const CheckoutContentInner = ({
               config={paywallConfig!}
             />
           )}
+          {(current.matches(CheckoutState.fiatLocks) ||
+            current.matches(CheckoutState.locks)) &&
+            !account.emailAddress &&
+            !!paywallConfig!.unlockUserAccounts && (
```
Comment (julien51) [inline unlock-app/src/components/content/CheckoutContent.tsx:164]: simple enough.

File: unlock-app/src/hooks/useFiatPurchaseKey.ts [hunk 2]
```diff
@@ -1,17 +1,47 @@
 import { useState, useEffect, useContext } from 'react'
```
Comment (julien51) [inline unlock-app/src/hooks/useFiatPurchaseKey.ts:1]: Thanks for renaming the file.

**Pull request merged**